src/components/KidsForm.js

```diff
@@ -0,0 +1,173 @@
+import styled from 'styled-components';
+import AddIcon from './icons/AddIcon.js';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { nanoid } from 'nanoid';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+export default function KidsForm({ kidsData, setKidsData, setShowMessage }) {
+  const [startDate, setStartDate] = useState(new Date());
+  const [nameInput, setNameInput] = useState('');
+  const [isName, setIsName] = useState(false);
+  const navigate = useNavigate();
+
+  const options = { year: 'numeric', month: 'numeric', day: 'numeric' };
+
+  const filteredName = kidsData.filter(item => item.name.toLowerCase().trim() === nameInput.toLowerCase().trim());
+  const filteredNameAndDate = kidsData.filter(
+    item =>
+      item.birthDate === startDate.toLocaleDateString('de-DE', options) &&
+      item.name.toLowerCase().trim() === nameInput.toLowerCase().trim()
+  );
+
+  const disabledButton = nameInput.length <= 1 || startDate === null || filteredNameAndDate.length > 0 ? true : false;
+
+  function handleTextInput(e) {
+    setNameInput(e.target.value);
+  }
```

Comment:
For better readability and because you are doing so in your submit function, I would recommend user `event` here instead of `e`.
```suggestion
  function handleTextInput(event) {
    setNameInput(event.target.value);
  }
```

Reply:
thx 👍